tests/basic/pool_function_test.go

```diff
@@ -216,16 +216,47 @@ var _ = Describe("{PoolExpandWithPXRestart}", func() {
 		EndTorpedoTest()
 	})
 
+	It("Restart PX after pool expansion", func() {
+		StartTorpedoTest("RestartAfterPoolExpansion",
+			"Restart PX after pool expansion", nil, testrailID)
+
+		Step("Select a pool that has I/O and expand it by 100 GiB with add-disk type. ", func() {
+			originalSizeInBytes = poolToResize.TotalSize
+			targetSizeInBytes = originalSizeInBytes + 100*units.GiB
+			targetSizeGiB = targetSizeInBytes / units.GiB
+			log.InfoD("Current Size of the pool %s is %d GiB. Trying to expand to %v GiB with type add-disk",
+				poolIDToResize, poolToResize.TotalSize/units.GiB, targetSizeGiB)
+			triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK)
+		})
+
+		Step("Wait for expansion to finish and restart PX", func() {
+			resizeErr := waitForOngoingPoolExpansionToComplete(poolIDToResize)
+			dash.VerifyFatal(resizeErr, nil, "Pool expansion does not result in error")
+			log.FailOnError(Inst().V.RestartDriver(*storageNode, nil),
+				fmt.Sprintf("Error restarting px on node [%s]", storageNode.Name))
+			log.FailOnError(Inst().V.WaitDriverUpOnNode(*storageNode, addDriveUpTimeOut),
+				fmt.Sprintf("Timed out waiting for px to come up on node [%s]", storageNode.Name))
+		})
+
+		Step("Ensure pool is up and running", func() {
+			// Ensure pool is up and running
+			poolToResize = getStoragePool(poolIDToResize)
+			// Ensure poolToResize is not nil
+			dash.VerifyFatal(poolToResize != nil, true, "Pool is up and running after restart")
+			verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, targetSizeGiB)
```

Comment:
Q. In what cases do we expect new size to be equal to existing size after an expand?

Reply:
Not sure I get your question.. Here we are checking the updated size is equal or larger than the "targetSize", not "existing size", wonder if you were confused.

Reply:
Thanks for clarifying.